client/keys.go

```diff
@@ -468,6 +470,21 @@ func (k *Key) CertDERBytes() []byte {
 	return k.cert.Raw
 }
 
+// SetCert assigns the provided certificate to the key after verifying it matches the key.
+func (k *Key) SetCert(cert *x509.Certificate) error {
```

Comment:
Nit: we should also be using this method internally for the various methods that set certificates, basically the 4 methods that call: `getCertificateFromNvram`.

Reply:
You may have to change that helper function to be a helper method, but it should be fine.